File: js/test/Exchange/test.fetchTrades.js
'use strict'

const assert = require ('assert');
const testTrade = require ('./test.trade.js');
const testSharedMethods = require ('./test.sharedMethods.js');

async function testFetchTrades (exchange, symbol) {
    const method = 'fetchTrades';
    const trades = await exchange[method] (symbol);
    assert (Array.isArray(trades), exchange.id + ' ' + method + ' ' + symbol + ' must return an array. ' + exchange.json(trades));
    console.log (exchange.id, symbol, 'fetched', trades.length, 'trades');
    const now = exchange.milliseconds ();
    for (let i = 0; i < trades.length; i++) {
        testTrade (exchange, method, trades[i], symbol, now);
    }
    testSharedMethods.reviseSortedTimestamps (exchange, method, symbol, trades);
}

module.exports = testFetchTrades;

File: julia/Ccxt/test/exchange/test.fetchTrades.jl
using Test
using Ccxt
function testFetchTrades(exchange, skippedProperties, symbol)

    method = "fetchTrades";
    trades = Base.fetch(fetchTrades(exchange, symbol));
    assertNonEmtpyArray(testSharedMethods, exchange, skippedProperties, method, trades);
    now = milliseconds(exchange);
    i = 0
    while functions.ccxtruthy(functions.ccxt_lt(i, length(trades)))
        testTrade(exchange, skippedProperties, method, get(trades, i + 1, nothing), symbol, now);
        assertInArray(testSharedMethods, exchange, skippedProperties, method, get(trades, i + 1, nothing), "takerOrMaker", ["taker", nothing]);
        i += 1
    end
    if functions.ccxtruthy(!functions.ccxtruthy((ccxt_in("timestampSort", skippedProperties))))
        assertTimestampOrder(testSharedMethods, exchange, method, symbol, trades);
    end
    return true
end
